barretenberg/cpp/src/barretenberg/dsl/acir_proofs/c_bind.cpp

```diff
@@ -19,11 +19,10 @@ WASM_EXPORT void acir_get_circuit_sizes(
     auto constraint_system =
         acir_format::circuit_buf_to_acir_format(from_buffer<std::vector<uint8_t>>(acir_vec), *honk_recursion);
     auto builder = acir_format::create_circuit(constraint_system, 1 << 19, {}, *honk_recursion);
-    *exact = htonl((uint32_t)builder.get_num_gates());
-    auto num_extra_gates = builder.get_num_gates_added_to_ensure_nonzero_polynomials();
-    auto total_with_extra_gates = builder.get_total_circuit_size() + num_extra_gates;
-    *total = htonl((uint32_t)total_with_extra_gates);
-    *subgroup = htonl((uint32_t)builder.get_circuit_subgroup_size(builder.get_total_circuit_size()));
+    builder.finalize_circuit(/*ensure_nonzero=*/true);
```

Comment:
we just overestimate if we ever have to pick. This function is used by plonk and honk, so we just go with adding more gates